### ISSUE TITLE
Emmet breaks custom html snippet expansion

### ISSUE THREAD

**Issue by serkandurusoy** [emmetio/emmet-atom#146]
As may be followed at https://github.com/ThusStyles/meteor-snippets/issues/4#issuecomment-57050554 emmet's snippet expansion with tab on html files breaks other custom snippets' tab expansion.
##

**Comment by ex-jedi** [comment]
Yep, having the same issue here. I'd like to find a way to disable expanding unknown tags. This is the one niggle I have that's keeping me from jumping off Sublime, which seems to handle this pretty well.

**Comment by kumarasinghe** [comment]
#### Alternatively...

You can make `autocomplete-plus` package show `snippets` by installing [autocomplete-snippets] ("https://atom.io/packages/autocomplete-snippets") package. In this way,
- You see snippets suggestions while you type.
- You can use emmet with tab.

Wow! | I | ♥ | Love | markdown! | `Cheers!!`
-------- | ----- | ----- | --------- | ---------- | -----------

**Comment by manolenso** [comment]
Thank's for sharing!

**Comment by MethodGrab** [comment]
If anyone is still having problems expanding HTML attribute snippets with `emmet-atom` + `autocomplete-plus` (Atom 0.199.0), my work around is to overwrite `enter` in my `keymap.cson`:
```
'.platform-darwin atom-pane atom-text-editor:not([mini])':
	'enter': 'autocomplete-plus:confirm'
```

This allows emmet expansions with `tab` and built-in expansions with `enter`.

**Comment by filipeaguiar** [comment]
It's still happening in 0.207.0. I've created 2 snippets to add the wiredep tags: wirecss and wirejs.
When I write `wirecss` I can see that the snippet is recognized by Atom, as it show the green arrow in the dropdown. But when I press **tab** I get `<wirecss></wirecss>` instead.
That's very annoying, to be honest.

**Comment by bjmiller** [comment]
The way to "fix" this is to edit the expand-abbreviation-with-tab command to check whether the current symbol left of the cursor is a valid emmet abbreviation, and call abortKeyBinding() if it isn't.  For reference: https://atom.io/docs/v1.0.0/behind-atom-keymaps-in-depth

**Comment by ethanclevenger91** [comment]
[autocomplete-snippets](https://atom.io/packages/autocomplete-snippets) works great for _starting_ a snippet, but if you have multiple tab stops, advancing will trigger Emmet again

**Comment by ethanclevenger91** [comment]
[Here's a solution](https://github.com/atom/autocomplete-plus/issues/86#issuecomment-82661027). You may need to either add the `text html php` grammar to the final keymap if you haven't [set up `php` to extend `html` context](https://github.com/emmetio/emmet-atom/issues/227#issuecomment-77691141) in Emmet

**Comment by sheriffderek** [comment]
Are there any updates on this? The goal is this:

- with emmet installed, tab completes the tag as expected with emmet
- if a snippet is used - then this overrides whatever emmet is doing and behaves as expected with snippets

Right? Sublime acts this way. Maybe some smarty-pants can look at their implementation?

**Comment by petrchutny** [comment]
I would also appreciate Emmet developers to enable an option in the setting, which would stop epanding non-existing words into pair tags and let the snippets expand by tab instead.

**Comment by sergeche** [comment]
You should try new Emmet 2 alpha instead: https://github.com/emmetio/atom-plugin
